From: Gerald Schaefer <geraldsc@de.ibm.com>
Subject: zfcp: erp failed status bit will not be set
References: bnc#482818,LTC#51962

Problem:     It will not be necessary to set the erp failed status bit
             in case a scsi device is removed by the scsi mid layer. 
             In the case a scsi device is unavailable for a short time
             (15 to 20 seconds) a FCP unit will not get on-line again.
Solution:    Do not change the status bits if slave destroy is called.

Acked-by: John Jolly <jjolly@suse.de>
---
 drivers/s390/scsi/zfcp_scsi.c |    1 -
 1 file changed, 1 deletion(-)

Index: linux/drivers/s390/scsi/zfcp_scsi.c
===================================================================
--- linux.orig/drivers/s390/scsi/zfcp_scsi.c
+++ linux/drivers/s390/scsi/zfcp_scsi.c
@@ -28,7 +28,6 @@ static void zfcp_scsi_slave_destroy(stru
 	struct zfcp_unit *unit = (struct zfcp_unit *) sdpnt->hostdata;
 	atomic_clear_mask(ZFCP_STATUS_UNIT_REGISTERED, &unit->status);
 	unit->device = NULL;
-	zfcp_erp_unit_failed(unit, 12, NULL);
 	zfcp_unit_put(unit);
 }
 
